pkg/fioapp/layers.go

```diff
@@ -167,11 +157,20 @@ func GetAppLayersFromMap(ctx context.Context, svcImages map[string]string, archL
 		// different images can consists of the same layers (layer intersection across images)
 		appLayers[arch] = make(map[string]distribution.Descriptor)
 		for manSvc, d := range manifests {
-			manifest, err := GetManifest(ctx, manSvc, d)
+			manifest, err := manSvc.Get(ctx, d)
 			if err != nil {
 				return nil, err
 			}
-			for _, layer := range manifest.Layers {
+			var layers []distribution.Descriptor
+			switch v := manifest.(type) {
```

Comment:
I'd move this `switch` to `GetManifest` just to encapsulate manifest fetching in on place.

Reply:
It's two different data types. I started to define an interface "LayeredManifest", but these are variables in the struct.

Reply:
ok, not important.